### PR TITLE
Add C implementation for `@stdlib/math/base/special/expm1rel`

### LINKED ISSUE #768: [RFC]: Add C implementation for `@stdlib/math/base/special/expm1rel`
### Description

This RFC proposes adding C implementation for [`@stdlib/math/base/special/expm1rel`](https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/math/base/special/expm1rel).

```c
double stdlib_base_expm1rel( const double x );
```

### Related Issues

None.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Pranavchiku**
Resolves #768.

## Checklist

- [x] update readme.md
- [x] include.gypi
- [x] binding.gyp
- [x] include/stdlib/math/base/special/
- [x] src
- [x] manifest.json
- [x] lib
- [x] examples
- [x] benchmark
- [x] test
* * *

@kgryte

### CODE REVIEW

**Comment by Pranavchiku** [comment]
All tests and benchmarks cleared, this PR can have a review.

File: lib/node_modules/@stdlib/constants/float64/eps/include/stdlib/constants/float64/eps.h [hunk 1]
```diff
@@ -22,6 +22,6 @@
 /**
 * Macro for the difference between one and the smallest value greater than one that can be represented as a double-precision floating-point number.
 */
-#define STDLIB_CONSTANT_FLOAT64_EPS 2.22044604925031308e-16;
+#define STDLIB_CONSTANT_FLOAT64_EPS 2.22044604925031308e-16
```
Comment (kgryte) [inline lib/node_modules/@stdlib/constants/float64/eps/include/stdlib/constants/float64/eps.h:25]: Thanks for catching this!

File: lib/node_modules/@stdlib/math/base/special/expm1rel/README.md [hunk 2]
```diff
@@ -114,6 +114,96 @@ for ( i = 0; i < 100; i++ ) {
...
+#include <stdlib.h>
+#include <stdio.h>
+
+int main() {
+    double x;
+    double v;
+    int i;
+    
+    for ( i = 0; i < 100; i++ ) {
+        x = ( ( ( double ) rand() / ( double ) RAND_MAX ) * 100.0 ) - 50.0;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/expm1rel/README.md]:
```suggestion
        x = ( ( (double)rand() / (double)RAND_MAX ) * 100.0 ) - 50.0;
```

File: lib/node_modules/@stdlib/math/base/special/expm1rel/examples/c/example.c [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+#include <stdlib.h>
+#include <stdio.h>
+
+int main() {
+    double x;
+    double v;
+    int i;
+    
+    for ( i = 0; i < 100; i++ ) {
+        x = ( ( ( double ) rand() / ( double ) RAND_MAX ) * 100.0 ) - 50.0;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/expm1rel/examples/c/example.c]:
```suggestion
        x = ( ( (double)rand() / (double)RAND_MAX ) * 100.0 ) - 50.0;
```

Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/expm1rel/examples/c/example.c]:
```suggestion
        x = ( ( (double)rand() / (double)RAND_MAX ) * 100.0 ) - 50.0;
```

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @Pranavchiku!

**Pull request merged**